modules/samples/firealarm/component/plugin/src/main/java/org/homeautomation/firealarm/plugin/impl/DeviceTypeManager.java

```diff
@@ -0,0 +1,259 @@
+import org.homeautomation.firealarm.plugin.exception.FirealarmPluginException;
+import org.homeautomation.firealarm.plugin.impl.dao.DeviceTypeDAO;
+import org.wso2.carbon.device.mgt.common.Device;
+import org.wso2.carbon.device.mgt.common.DeviceIdentifier;
+import org.wso2.carbon.device.mgt.common.DeviceManagementException;
+import org.wso2.carbon.device.mgt.common.DeviceManager;
+import org.wso2.carbon.device.mgt.common.EnrolmentInfo;
+import org.wso2.carbon.device.mgt.common.FeatureManager;
+import org.wso2.carbon.device.mgt.common.configuration.mgt.TenantConfiguration;
+import org.wso2.carbon.device.mgt.common.license.mgt.License;
+import org.wso2.carbon.device.mgt.common.license.mgt.LicenseManagementException;
+
+import java.util.List;
+
+/**
+ *
+ */
+public class DeviceTypeManager implements DeviceManager {
+
+    private static final Log log = LogFactory.getLog(DeviceTypeManager.class);
+
+    private static final DeviceTypeDAO connectedCupDAO = new DeviceTypeDAO();
+
+    @Override
+    public FeatureManager getFeatureManager() {
+        return null;
+    }
+
+    @Override
+    public boolean saveConfiguration(TenantConfiguration tenantConfiguration)
+            throws DeviceManagementException {
+        //TODO implement this
+        return false;
+    }
+
+    @Override
+    public TenantConfiguration getConfiguration() throws DeviceManagementException {
+        //TODO implement this
+        return null;
+    }
+
+    @Override
+    public boolean enrollDevice(Device device) throws DeviceManagementException {
+        boolean status;
+        try {
+            if (log.isDebugEnabled()) {
+                log.debug("Enrolling a new Connected Cup device : " + device.getDeviceIdentifier());
+            }
+            DeviceTypeDAO.beginTransaction();
+            status = connectedCupDAO.getConnectedCupDeviceDAO().addDevice(device);
+            DeviceTypeDAO.commitTransaction();
+        } catch (FirealarmPluginException e) {
+            try {
+                DeviceTypeDAO.rollbackTransaction();
+            } catch (FirealarmPluginException iotDAOEx) {
+                String msg = "Error occurred while roll back the device enrol transaction :" + device.toString();
+                log.warn(msg, iotDAOEx);
+            }
+            String msg = "Error while enrolling the Connected Cup device : " + device.getDeviceIdentifier();
+            log.error(msg, e);
+            throw new DeviceManagementException(msg, e);
+        }
+        return status;
+    }
+
+    @Override
+    public boolean modifyEnrollment(Device device) throws DeviceManagementException {
+        boolean status;
+        try {
+            if (log.isDebugEnabled()) {
+                log.debug("Modifying the Connected Cup device enrollment data");
+            }
+            DeviceTypeDAO.beginTransaction();
+            status = connectedCupDAO.getConnectedCupDeviceDAO().updateDevice(device);
+            DeviceTypeDAO.commitTransaction();
```

Comment:
There is a possibility for an exception at the time of beginning or committing a transaction. Why aren't we considering those possibilities here?